apps/api/src/database/repositories/leaderboard-repository.ts

```diff
@@ -348,15 +278,25 @@ async function getBulkAgentMetricsImpl(agentIds: string[]): Promise<
       )
       .groupBy(competitionsLeaderboard.agentId);
 
+    // Query 8: Get all agent scores for rank calculation in service layer
+    const allAgentScoresQuery = dbRead
```

Comment:
We are loading every agent score row here.  Is that ok?
It looks like we only have 44 scored agents currently, but we will in theory have 100,000+ agents in the future right?

Reply:
since the original logic technically does this, i created a follow up https://linear.app/recall-labs/issue/APP-202/optimize-and-limit-paginate-getagentmetrics-logic. i think we'll want to address this before the "always weekly comps" happens.